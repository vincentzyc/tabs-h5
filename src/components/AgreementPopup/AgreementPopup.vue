<template>
  <van-popup v-model:show="show" class="agreement-popup" style="max-width: 576px" teleport="body">
    <div class="agreement-wrap">
      <iframe :src="text" frameborder="0" style="width: 100%; height: 100%"></iframe>
    </div>
    <div class="agreement-confirm" @click="close()">我知道了</div>
  </van-popup>
</template>

<script setup lang="ts">
defineProps<{
  show: boolean;
  text: string;
}>();
const emit = defineEmits<{
  (e: "update:show", bool: boolean): void;
}>();
function close() {
  emit("update:show", false);
}
</script>
