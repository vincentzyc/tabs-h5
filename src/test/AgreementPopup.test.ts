import { mount } from '@vue/test-utils'
import AgreementPopup from '@/components/AgreementPopup/AgreementPopup.vue'

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

test('mount AgreementPopup component', async () => {
  expect(AgreementPopup).toBeTruthy()

  const wrapper = mount(AgreementPopup, {
    props: {
      text: "https://fanyi.baidu.com/",
      show: true,
    },
  })

  await nextTick();

  await sleep(2000);

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('.agreement-confirm').trigger('click')

  const emitEvent = wrapper.emitted('update:show')

  expect(emitEvent).toHaveLength(1)
})
