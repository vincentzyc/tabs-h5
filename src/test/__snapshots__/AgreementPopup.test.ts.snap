// Vitest Snapshot v1

exports[`mount AgreementPopup component 1`] = `
"<!--teleport start-->
<!--teleport end-->"
`;
